File: Greedy/3075. Maximize Happiness of Selected Children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        happiness.reverse()

        ans, used = 0, 0
        for happy in happiness:
            if used == k or happy - used <= 0:
                break
            ans += happy - used
            happy += 1
            used += 1
        return ans

File: 3075. Maximize Happiness of Selected Children.cpp

class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) 
    {
        priority_queue<long long>pq;
        for(int v:happiness) pq.push(v);
        
        int i = 0;
        long long ans = 0;
        while(k--)
        {
            long long cur = pq.top() - i;
            if(cur <= 0) break;
            ans += cur;
            pq.pop();
            i++;
        }
        return ans;
    }
};